android/app/src/main/java/com/microsoft/codepush/react/CodePushNativeModule.java

```diff
@@ -79,24 +79,26 @@ private boolean isReactApplication(Context context) {
         return false;
     }
 
-    private void loadBundleLegacy() {
-        Activity currentActivity = getCurrentActivity();
-        Intent intent = currentActivity.getIntent();
-        currentActivity.finish();
-        currentActivity.startActivity(intent);
-
+    private void loadBundleLegacy(final Activity currentActivity) {
         mCodePush.invalidateCurrentInstance();
+
+        currentActivity.runOnUiThread(new Runnable() {
```

Comment:
Almost all of loadBundle() is inside a `try ...catch` which would catch NPEs arising from a null currentActivity, and then fall back to this legacy behaviour. I'm just wondering whether there is a chance that at the top of the `loadBundle` method where we try to get the current activity, it is already null, and we would be handing over this `null` instance to `loadBundleLegacy`. I thought maybe we should do this check just because most other RN native modules also do the same check. But that said, there is no evidence we have ever seen this happen in practice, so I am fine with you merging this as is.